lib/promoted/ruby/client/request_builder.rb

```diff
@@ -92,8 +98,16 @@ def log_request_params(include_insertions: true, include_request: true)
             cohort_membership: @experiment,
             client_info: @client_info
           }
+
+          request[:request_id] = @id_generator.newID if not request[:request_id]
+
+          # Log request allows for multiple requests but here we only send one.
           params[:request] = [request] if include_request
-          params[:insertion] = compact_metrics_insertions if include_insertions
+
+          if include_insertions
+            params[:insertion] = compact_metrics_insertions if include_insertions
+            add_missing_ids_on_insertions! request, params[:insertion]
```

Comment:
@danbosnichill note that regardless of custom compact function behavior, we'll still add the ids and the user cannot override it.